### PR TITLE
Include generated cabal files

### PR DESCRIPTION
**Pull request by snoyberg**
Note: Documentation fixes for https://docs.haskellstack.org/en/stable/ should target the "stable" branch, not master.

Please include the following checklist in your PR:

* [X] Any changes that could be relevant to users have been recorded in the ChangeLog.md
* [X] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -26,7 +26,8 @@ extra-deps:
 - http-download-0.2.0.0@rev:0
 - github: snoyberg/filelock
   commit: 97e83ecc133cd60a99df8e1fa5a3c2739ad007dc
-- pantry-0.3.0.0@rev:0
+- github: commercialhaskell/pantry
+  commit: 7ac9f9364bf8491567f3d00d948bbf9c84cca5b8
```
Comment (qrilka) [inline stack.yaml]: What's the motivation for not using a proper version from Hackage?

File: ChangeLog.md [hunk 2]
```diff
@@ -23,10 +23,17 @@ Major changes:
 
 * Fully remove GHCJS support.
 
+* Remove the `freeze` command. It has been replaced by lock files.
```
Comment (qrilka) [inline ChangeLog.md]: Is this somehow related to generated cabal files?

File: stack.yaml (hunk 1 above)
Comment (snoyberg) [reply stack.yaml]: I wanted to test this PR and get review before committing to a release. With your review, I'll go ahead and release + update this PR.

File: ChangeLog.md (hunk 2 above)
Comment (snoyberg) [reply ChangeLog.md]: Yes, I explained it in a commit message, but didn't want to waste space in the changelog. Since there's a change in the desired semantics of freeze, we could either introduce that change, _or_ just get rid of the command. Since we want to get rid of the command anyway, may as well just do it now.

Comment (qrilka) [reply ChangeLog.md]: Thanks
